.github/workflows/test-build-run.yml

```diff
@@ -162,12 +161,15 @@ jobs:
 
     - name: Build raft
       run: |
+        shopt -s globstar
         cd raft
         git log -n 1
-        autoreconf -i
-        ./configure --enable-debug --enable-backtrace
-        make -j4
-        sudo make install
+        cc -pthread -g3 -shared -fPIC \
+          -D_GNU_SOURCE -DLZ4_AVAILABLE -DLZ4_ENABLED -DHAVE_LINUX_AIO_ABI_H -DRAFT_ASSERT_WITH_BACKTRACE \
+          -o libraft.so src/**/*.c -lbacktrace -luv -llz4
+        sudo install -m755 libraft.so /usr/local/lib
+        sudo install -m644 include/raft.h /usr/local/include
+        sudo install -m644 -D -t /usr/local/include/raft include/raft/*.h
```

Comment:
How much speed up does this provide?

Is it worth the custom code? This setup code can break and/or need maintenance, in case details like needed libraries and file location change.

I presume the Jepsen suite takes a long time anyway? Maybe there are other more effective ways to reduce the total time the job takes?

Reply:
> So I'll lay my cards on the table and say that for raft and dqlite I don't believe in abstracting the build process in cases like this, where there are no platform quirks to worry about. Fundamentally, each project is a pile of .c files that can be compiled separately and then linked or archived into a final artifact. You have to pass -pthread and -D_GNU_SOURCE, because we use those features in our code, and you have to pass certain other -D flags to enable optional features (snapshot compression, backtraces); but there's nothing really complicated going on, and when targeting a known platform Autotools is not adding much value, and I'd much rather use cc directly. 

That feels a separate conversation than this PR though.

> (Passing -D and -l flags to cc is not really any different from passing --enable-* flags to ./configure, and ultimately I think in the language of cc flags.)
>

Same as above, probably better to talk about this separately. I'll just note that `./configure` options are supposed to be a bit more friendly. At least I'd personally find that easier than having to figure out flags.

> As for forgetting the existence of this separate build step, I already bypass the build system when hacking on raft and dqlite locally, so I'd have to update that script if the build process changed. I can update Jepsen at the same time.

Sure, but there are also other devs too.

There are also a number of flags that we normally pass to cc when using autotools that I don't think we're passing here.

My point is mainly that unless there's a specific reason to **not** use the same regular build process that users and packagers are going to use, I think Jepsen tests should stick to that regular build process (whatever it is, it's not set in stone), so we minimize the risk for subtle differences.

I'm not totally sure the speed-up of this change is something critical enough to give up consistency.